src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,45 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health = 100, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    if (this.hidden === false) {
+      this.hidden = true;
+    }
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+
+    Animal.alive.push(this);
+  }
+
+  bite(victim) {
+    if (victim instanceof Herbivore && victim.hidden === false) {
+      victim.health -= 50;
+
+      if (victim.health === 0) {
+        Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
```

Comment:
[CODE PERFORMANCE] - It is unnecessary to check if the victim's health is exactly `0` before filtering the `Animal.alive` array. You can remove this check and filter the array directly since the `filter` method will handle all cases where health is less than or equal to `0` efficiently.

Reply:
I want to do filtering only in cases when somebody was killed for saving performance